Sources/Subscription/Managers/StorePurchaseManager.swift

```diff
@@ -129,11 +135,21 @@ public final class DefaultStorePurchaseManager: ObservableObject, StorePurchaseM
         Logger.subscription.info("[StorePurchaseManager] updateAvailableProducts")
 
         do {
-            let availableProducts = try await Product.products(for: productIdentifiers)
-            Logger.subscription.info("[StorePurchaseManager] updateAvailableProducts fetched \(availableProducts.count) products")
+            let currentStorefrontCountryCode = await Storefront.current?.countryCode ?? ""
+            self.currentStorefrontRegion = SubscriptionRegion.matchingRegion(for: currentStorefrontCountryCode) ?? .usa
```

Comment:
Can defaulting to `.usa` here cause problems for anyone outside of the US? Seems to me like we should **always** have a matching region here, should we at least add a pixel for it?